pages/details/_title.vue

```diff
@@ -3,46 +3,60 @@
     <v-card elevation="2">
       <v-img
         class="white--text align-end"
-        height="200px"
-        :src="singleData.urlToImage"
+        :src="product.urlToImage"
       >
-        <v-card-title>{{ singleData.title }}</v-card-title>
+        <v-card-title>{{ product.title }}</v-card-title>
       </v-img>
 
       <v-card-subtitle class="pb-0">
-        {{ singleData.description }}
+        {{ product.description }}
       </v-card-subtitle>
 
-      <v-card-text class="text--primary">{{ singleData.author }} </v-card-text>
+      <v-card-text class="text--primary">{{ product.author }} </v-card-text>
       <h5 class="date-v">Published date : 20th January 2023</h5>
     </v-card>
   </div>
 </template>
-<script>
+
+<script lang="ts">
 import axios from 'axios'
-export default {
+import { Dictionary } from 'vue-router/types/router'
+import Vue from 'vue'
+// import Vue from 'vue/types/umd'
+import {SingleDataType} from '../../interface/productInterface'
+export default Vue.extend ({
   name: 'details',
   data() {
+    const data: Array<SingleDataType> = []
+    let product: SingleDataType = {
+      description: '',
+      urlToImage: '',
+      title: '',
+      author: ''
+    }
+
     return {
-      data: [],
-      singleData: [],
+      data,
+      product,
     }
   },
-  async mounted() {
+  mounted() {
     this.fetchData(this.$route.params)
   },
   methods: {
-    async fetchData(params) {
+    async fetchData(params: Dictionary<String>): Promise<void> {
```

Comment:
same here 
what's the use of await here

Reply:
Resolved